src/ReactLeafletMap/layers/VectorLayer/index.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+
+import CanvasVectorLayer from "../CanvasVectorLayer"
+import DiagramLayer from "../DiagramLayer";
+import IndexedVectorLayer from "../IndexedVectorLayer";
+import SvgVectorLayer from "../SvgVectorLayer";
+import TiledVectorLayer from "../TiledVectorLayer";
+
+class VectorLayer extends React.PureComponent {
+	static propTypes = {
+		layerKey: PropTypes.string,
+		uniqueLayerKey: PropTypes.oneOfType([
+			PropTypes.number,
+			PropTypes.string
+		]),
+		onClick: PropTypes.func,
+		opacity: PropTypes.number,
+		options: PropTypes.object,
+		type: PropTypes.string,
+		view: PropTypes.object,
+		zoom: PropTypes.number,
+		zIndex: PropTypes.number
+	};
+
+	constructor(props) {
+		super(props);
+	}
+
+	getOptions() {
+		const props = this.props;
+		const renderAs = props.options?.renderAs;
+		let options = props.options;
+
+		if (renderAs) {
+			const boxRange  = props.view?.boxRange;
+			const renderAsData = _.find(renderAs, (renderAsItem) => {
+				const boxRangeRange = renderAsItem.boxRangeRange;
+
+				// Current boxRange is in defined range
+				return (boxRange > boxRangeRange[0] && boxRange <= boxRangeRange[1]) || (!boxRangeRange[0] && boxRange <= boxRangeRange[1]) || (boxRange > boxRangeRange[0] && !boxRangeRange[1]);
```

Comment:
Do we have some helper for it?

Reply:
You mean for line 42?

Reply:
yes

Reply:
55672d5f0a797875a6b75c07714f451d1ac8b6cb